provider/maas/environ.go

```diff
@@ -948,6 +951,33 @@ func (environ *maasEnviron) StartInstance(args environs.StartInstanceParams) (
 			return nil, errors.Annotatef(err, "cannot run instances")
 		}
 	}
+	done := make(chan struct{})
+	defer close(done)
+	go func(done chan struct{}) {
```

Comment:
Please do a few things here:

1. This method is *huge*. Please pull this goroutine out into its own function
2. This is a retryable operation; please use github.com/juju/rety.
3. Consolidate the function's conditionals; as we discussed, `args.StatusCallback` doesn't change once passed in. Create a closure somewhere that is a noop if this is nil. This allows you to collapse the conditional in your loop.
4. Please document why you're doing this and specifically call out that it's OK if this goroutine is never run.